demisto_sdk/commands/validate/validators/RM_validators/RM108_check_image_path.py

```diff
@@ -0,0 +1,117 @@
+from __future__ import annotations
+
+import re
+from abc import ABC
+from typing import Iterable, List
+
+from demisto_sdk.commands.common.constants import (
+    DOC_FILE_IMAGE_REGEX,
+    HTML_IMAGE_LINK_REGEX,
+    URL_IMAGE_LINK_REGEX,
+)
+from demisto_sdk.commands.content_graph.objects.content_item import ContentItem
+from demisto_sdk.commands.content_graph.parsers.related_files import RelatedFileType
+from demisto_sdk.commands.validate.validators.base_validator import (
+    BaseValidator,
+    ValidationResult,
+)
+
+ContentTypes = ContentItem
+
+
+class ImagePathValidator(BaseValidator, ABC):
+    error_code = "RM108"
+    description = (
+        "This validation verifies that images in the readme and description files are"
+        " relative and stored in doc_files."
+    )
+    rationale = (
+        "Using relative references to files in the repo folder enhances security by reducing reliance"
+        " on external links, minimizing the risk of link manipulation or redirection attacks. "
+    )
+    error_message = (
+        "{}. See https://xsoar.pan.dev/docs/integrations/integration-docs#images for further info on"
+        " how to add images to pack markdown files."
+    )
+    related_file_type = [RelatedFileType.README, RelatedFileType.DESCRIPTION_File]
+
+    def is_valid(self, content_items: Iterable[ContentTypes]) -> List[ValidationResult]:
+        return [
+            ValidationResult(
+                validator=self,
+                message=self.error_message.format(error_message),
+                content_object=content_item,
+            )
+            for content_item in content_items
+            if (error_message := self.validate_content_items(content_item))
+        ]
+
+    def validate_content_items(self, content_item):
+        """Check if the content items are valid by passing verify_absolute_images_not_exist and verify_relative_saved_in_doc_files.
+
+        Arguments:
+            content_item {ContentTypes} -- The content item to check.
+
+        Returns:
+            str -- The error message if the content item isn't valid.
+        """
+        raise NotImplementedError("Subclasses must implement this method.")
+
+    def verify_absolute_images_not_exist(self, content_item) -> str:
+        """Check if the content item contains exists image paths.
+
+        Arguments:
+            content_item {ContentTypes} -- The content item to check.
+
+        Returns:
+            str -- The error message if the content item contains absolute paths.
+        """
+        absolute_links = re.findall(
+            URL_IMAGE_LINK_REGEX + r"|" + HTML_IMAGE_LINK_REGEX,
+            content_item,
+            re.IGNORECASE | re.MULTILINE,
+        )
+        if absolute_links:
+            absolute_links = [
+                absolute_link[1] if absolute_link[0] else absolute_link[2]
```

Comment:
What is this meant for?

Reply:
searching with the regex returns each link group  ("image]", "link" l)not only with the link itself.  Since MD file has the same structure for images we can use this function to get the link itself